huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_directory.go

```diff
@@ -0,0 +1,332 @@
+				ForceNew: true,
+			},
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"parent_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"root_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"children": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"directory_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"qualified_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceStudioDirectoryCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	workspaceID := d.Get("workspace_id").(string)
+	dirType := d.Get("type").(string)
+
+	//nolint:misspell
+	createDirectoryHttpUrl := "v2/{project_id}/design/directorys"
+	createDirectoryProduct := "dataartsv2"
+
+	createDirectoryClient, err := cfg.NewServiceClient(createDirectoryProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio V2 Client: %s", err)
+	}
+	createDirectoryPath := createDirectoryClient.Endpoint + createDirectoryHttpUrl
+	createDirectoryPath = strings.ReplaceAll(createDirectoryPath, "{project_id}", createDirectoryClient.ProjectID)
+
+	createDirectoryOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": workspaceID},
+	}
+	createDirectoryOpt.JSONBody = utils.RemoveNil(buildCreateDirectoryBodyParams(d))
+	createDirectoryResp, err := createDirectoryClient.Request("POST", createDirectoryPath, &createDirectoryOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createDirectoryRespBody, err := utils.FlattenResponse(createDirectoryResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	directory := utils.PathSearch("data.value", createDirectoryRespBody, nil)
+
+	id, err := jmespath.Search("id", directory)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio directory: %s is not found in API response", id)
+	}
+	qualifiedName, err := jmespath.Search("qualified_name", directory)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio directory: %s is not found in API response", qualifiedName)
+	}
+	d.SetId(workspaceID + "/" + dirType + "/" + qualifiedName.(string))
```

Comment:
as we discussed before, we can set the ID with d.SetId(id), and use `<workspace_id>/<directory_type>/<directory_path>` to import it